### PR TITLE
Yet another cleanup PR

### PR DESCRIPTION
**Pull request by bmarty**
- use invalidate() from MvRx
- use throttleFirst instead of debounce
- remove useless observe on main thread
- remove useless calls to super.invalidate()

### CODE REVIEW

File: vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt [hunk 1]
```diff
@@ -48,8 +43,7 @@ class SharedSecuredStorageKeyFragment @Inject constructor(
         ssss_restore_with_key_text.text = getString(R.string.enter_secret_storage_input_key)
 
         ssss_key_enter_edittext.editorActionEvents()
-                .debounce(300, TimeUnit.MILLISECONDS)
-                .observeOn(AndroidSchedulers.mainThread())
+                .throttleFirst(300, TimeUnit.MILLISECONDS)
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt:46]: Why do you replace debounce by throttle? In the context of an edittext it's better to keep with debounce

File: vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt [hunk 2]
```diff
@@ -48,8 +43,7 @@ class SharedSecuredStorageKeyFragment @Inject constructor(
         ssss_restore_with_key_text.text = getString(R.string.enter_secret_storage_input_key)
 
         ssss_key_enter_edittext.editorActionEvents()
-                .debounce(300, TimeUnit.MILLISECONDS)
-                .observeOn(AndroidSchedulers.mainThread())
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt:52]: Also, why removing mainThread? the submit method is changing some ui, we might get into trouble if we are not switching back to the ui thread

File: vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt:46]: We do not observe textChange here, but editorActionEvents(), to get EditorInfo.IME_ACTION_DONE event. So I think throttleFirst is better.

File: vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt:52]: we are in the main thread, editorActionEvents() is supposed to emit only on main thread (no?) and debounce does not change the thread, so I think observeOn(AndroidSchedulers.mainThread()) is not necesarry. But maybe I'm wrong?

Comment (ganfra) [reply vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt:52]: Deboune and throttle does change thread by default

File: vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt (hunk 1 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt:46]: Oh yes, didn't see that

File: vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStorageKeyFragment.kt:52]: Ok, I reverted this change

**Pull request merged**